lib/appmap/util.rb

```diff
@@ -21,6 +21,16 @@ module Util
     WHITE   = "\e[37m"
 
     class << self
+      def parse_function_name(name)
+        package_tokens = name.split('/')
+
+        class_and_name = package_tokens.pop
+        class_name, function_name, static = class_and_name.include?('.') ? class_and_name.split('.', 2) + [ true ] : class_and_name.split('#', 2) + [ false ]
+
+        raise "Malformed fully-qualified function name #{name}" unless function_name
+        [ package_tokens.empty? ? 'ruby' : package_tokens.join('/'), class_name, static, function_name ]
```

Comment:
Why `'ruby'`? It's a bit unexpected. Could it be `nil`?

Reply:
I am worried that if I do that I will break something else downstream. There hasn't been a situation where package could be nil, so I don't see a compelling reason to allow it now.